.github/workflows/tests.yml

```diff
@@ -70,3 +70,53 @@ jobs:
         with:
           name: test_results_${{ matrix.api-versions }}
           path: '**/build/test-results/**/TEST-*.xml'
+
+  instrumetation-tests:
+    runs-on: macos-latest
+    timeout-minutes: 60
+    needs: build
+
+    strategy:
+      # Allow tests to continue on other devices if they fail on one device.
+      fail-fast: false
+      matrix:
+        api-level: [ 26, 28, 29, 30 ]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK 11.0.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11.0.8
+
+      # Determine what emulator image to use. We run all API 29+ emulators using
+      # the google_apis image
+      - name: Determine emulator target
+        id: determine-target
+        env:
+          API_LEVEL: ${{ matrix.api-level }}
+        run: |
+          TARGET="default"
+          if [ "$API_LEVEL" -ge "26" ]; then
+            TARGET="google_apis"
+          fi
+          echo "::set-output name=TARGET::$TARGET"
+
+      - name: Run device tests
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: ${{ matrix.api-level }}
+          target: ${{ steps.determine-target.outputs.TARGET }}
+          profile: Galaxy Nexus
```

Comment:
Thanks. I will update it.

Reply:
> It looks like this isn't being defined for the ctesque tests

I have encountered a failed test(`BitmapFactoryTest#decodeResource`) affected by device density and size. Contributors, likes me, maybe run it locally with custom Android emulator, and it can cause this test failed again, so I added a commit to fix it.

Reply:
Done.